examples/pytorch/speech-recognition/run_speech_recognition_seq2seq.py

```diff
@@ -241,6 +246,9 @@ def __call__(self, features: List[Dict[str, Union[List[int], torch.Tensor]]]) ->
 
         batch = self.processor.feature_extractor.pad(input_features, return_tensors="pt")
 
+        if self.return_attention_mask:
```

Comment:
This is potentially confusing since with Wav2Vec2 models, we'll return an attention mask when we call `feature_extractor.pad`:
```python
batch = self.processor.feature_extractor.pad(input_features, return_tensors="pt")  # pad returns `input_values` and `attention_mask`
```
We can actually toggle whether this method returns an attention mask as well:
```python
batch = self.processor.feature_extractor.pad(input_features, return_tensors="pt", return_attention_mask=False)  # pad returns `input_values` now
```

But with Whisper, we'll only get an attention mask if it's in `features`:
```python
batch["attention_mask"] = torch.LongTensor([feature["attention_mask"] for feature in features])
```

I can't think of a clean way of distinguishing the two other than perhaps a subtle name change:
```suggestion
        if self.forward_attention_mask:
```

Reply:
Thanks for pointing this out and for the explication! But IMO wav2vec2 models are usually used in CTC architecture but not in run_speech_recognition_seq2seq.py no ?

Reply:
We actually first added this script based on the results of this paper: https://arxiv.org/abs/2111.09296

Here, they create a sequence-to-sequence ASR model by initialising an encoder with a Wav2Vec2 model and adding a transformer decoder on top. We extended this further by initialising the encoder with Wav2Vec2 and the transformer decoder with a BART decoder: https://github.com/huggingface/transformers/tree/main/examples/pytorch/speech-recognition#warm-started-speech-encoder-decoder-model

Reply:
Thanks again for the explication! Done in https://github.com/huggingface/transformers/pull/21942/commits/19c7a24b57a0e05910326780257d058ee8965fbf